frontend/catalyst/qfunc.py

```diff
@@ -202,13 +237,18 @@ def processing_fn(results):
         return dynamic_one_shot_partial(qnode)
 
     single_shot_qnode = transform_to_single_shot(qnode)
+    if mcm_config is not None:
+        single_shot_qnode.execute_kwargs["mcm_config"] = mcm_config
+    single_shot_qnode._dynamic_one_shot_called = True
     dev = qnode.device
-    single_shot = 1 if isinstance(dev, qml.devices.LegacyDevice) else qml.measurements.Shots(1)
-    single_name = dev.short_name if isinstance(dev, qml.devices.LegacyDevice) else dev.name
-    single_shot_qnode.device = qml.device(single_name, wires=dev.wires, shots=single_shot)
-    total_shots = (
-        dev.shots if isinstance(qnode.device, qml.devices.LegacyDevice) else dev.shots.total_shots
-    )
+    total_shots = get_device_shots(dev)
+
+    new_dev = copy(dev)
+    if isinstance(new_dev, qml.devices.LegacyDevice):
+        new_dev.shots = 1  # pragma: no cover
+    else:
+        new_dev._shots = qml.measurements.Shots(1)
+    single_shot_qnode.device = new_dev
```

Comment:
This is the uncovered line, so I've added a `# pragma: no cover` for now. I could also fix this by doing the following, but it is much less readable and honestly just bad.

```suggestion
    new_shots = 1 if instance(new_dev, qml.devices.LegacyDevice) else qml.measurements.Shots(1)
    setattr(new_dev, "shots" if isinstance(new_dev, qml.devices.LegacyDevice) else "_shots", new_shots)
```

Reply:
I think it makes sense that it's uncovered since we perform code coverage with a legacy device anymore since lightning upgraded.